src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java

```diff
@@ -187,6 +187,25 @@ public void testSupplementaryReadUnmappedMate() {
         }
     }
 
+    @Test
+    public void testHashCollisionHandling() {
+        File output = createTempFile("supplementaryReadUnmappedMate", "bam");
```

Comment:
I think this test is out of date now.

Reply:
Could you add tests that generate some keys from the different reads and show that they are not equal for non-equal reads?  Nothing too fancy.